xscen/diagnostics.py

```diff
@@ -262,11 +262,17 @@ def _message():
             for method, kwargs in missing.items():
                 kwargs.setdefault("freq", "YS")
                 for v in ds.data_vars:
-                    ms = getattr(xc.core.missing, method)(ds[v], **kwargs)
-                    if ms.any():
-                        _error(
-                            f"The variable '{v}' has missing values according to the '{method}' method.",
-                            "missing",
+                    if "time" in ds[v].dims:
+                        ms = getattr(xc.core.missing, method)(ds[v], **kwargs)
+                        if ms.any():
+                            _error(
+                                f"The variable '{v}' has missing values according to the '{method}' method.",
+                                "missing",
+                            )
+                    else:
+                        warnings.warn(
+                            f"Variable '{v}' has no time dimension. The missing data check will be skipped.",
+                            UserWarning,
```

Comment:
Feels like this could rather be an INFO logging call ?
Or you think it is abnormal that the user requested a missing check when there is no time dimension ?

Reply:
OK to change that to a logging call. The issue here is that it loops through all variables in the dataset, so if a single once has no `time` dimension, it crashes.